articles/how-to-use-local-storage-with-JavaScript/index.md

```diff
@@ -0,0 +1,248 @@
+          </li>
+        </ul>
+      </div>
+    </nav>
+  </header>
+
+   
+  <br>
+  <br>
+  <div class="container" id="formDiv">
+        <form class="form-signin" id="carForm">
+            <h1>Please sign in</h1>
+            <label for="carBrand" class="sr-only">Car</label>
+            <input type="text" id="carBrand" class="form-control" placeholder="Car brand" required autofocus><br>
+            <label for="carPrice" class="sr-only">Price</label>
+            <input type="text" id="carPrice" class="form-control" placeholder="Price" required><br>
+            <button class="btn btn-lg btn-primary btn-block" type="submit">Submit</button>
+        </form>
+        <br>
+        <br>
+
+        <button class="btn btn-lg btn-primary btn-block" id="retrieveButton">Retrieve records</button>
+        <br>
+        <div id="retrieve"></div>
+        <br>
+
+        <button class="btn btn-lg btn-danger btn-block" id="removeButton">Remove record</button>
+        <br>
+        <br>
+        
+        <button class="btn btn-lg btn-danger btn-block" id="clearButton">Clear all records</button>
+      
+  </div>
+
+</body>
+</html>
+
+```
+
+When the `submit` button is clicked, it takes the user input and passes it to the `store` function in the main.js file.
+
+Similarly, the `retrieveButton` will invoke the retrieveRecords function when clicked.
+
+removeButton invokes `removeItem()`.
+
+clearButton calls the `clearStorage()`.
+
+Here is the `main.js` file with the required functions.
+
+```javascript
+function store(){ //stores items in the local storage
+        var brand = document.getElementById('carBrand').value;
+        var price = document.getElementById('carPrice').value;
+    
+        const car = {
+            brand: brand,
+            price: price,
+        }
+        
+        window.localStorage.setItem('car',JSON.stringify(car));  
+        //converting object to string
+}
+
+function retrieveRecords(){ //retrieves items in the localStorage
+    console.log("retrive records");
+    var records = window.localStorage.getItem('car');
+    var paragraph = document.createElement("p");
+    var infor = document.createTextNode(records);
+    paragraph.appendChild(infor);
+    var element = document.getElementById("retrieve");
+    element.appendChild(paragraph);
+}
+
+
+function removeItem(){//deletes item from localStorage
+    localStorage.removeItem('car')
+    console.log("remove items");
+}
+
+function clearStorage(){ //clears the entire localStorage
+    localStorage.clear()
+    console.log("clear records");
+}
+
+window.onload =function(){ //ensues the page is loaded before functions are executed.
+    document.getElementById("carForm").onsubmit = store
+
+    document.getElementById("clearButton").onclick = clearStorage
+
+    document.getElementById("removeButton").onclick = removeItem
+
+    document.getElementById("retrieveButton").onclick = retrieveRecords
+
+}
+
+```
```

Comment:
can you briefly explain the other methods that you are using like document.createElement and appendChild etc.

Break the four functions into separate code snippets and briefly explain about them.

Reply:
Revised